experimental/incremental/task.go

```diff
@@ -43,6 +58,13 @@ type Task struct {
 
 	// Intrusive linked list node for cycle detection.
 	path path
+
+	// Set if we're currently holding the executor's semaphore. This exists to
+	// ensure that we do not violate concurrency assumptions, and is never
+	// itself mutated concurrently.
+	holding bool
+	// True if this task is executing on the g that called [Run].
```

Comment:
This feels like it could be cryptic for some readers. Maybe instead:
```suggestion
	// True if this task is a root task and executes on the goroutine that called [Run].
```